### PR TITLE
Upgrade python to 3.10

### PR DESCRIPTION
**Pull request by dmannarino**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): 
Tech debt

## What is the current behavior?

We're using Python 3.8

Issue Number: GTC-1994

## What is the new behavior?

-Python 3.10 is used

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,12 @@
-FROM tiangolo/uvicorn-gunicorn-fastapi:python3.8-slim
+FROM tiangolo/uvicorn-gunicorn-fastapi:python3.10-slim
 
 # Optional build argument for different environments
 ARG ENV
 
 RUN apt-get update -y \
-    && apt-get install --no-install-recommends -y gcc libc-dev musl-dev postgresql-client libpq-dev
+    && apt-get install --no-install-recommends -y gcc libc-dev musl-dev postgresql-client libpq-dev make
```
Comment (gtempus) [inline Dockerfile:7]: Why did we have to add `make`?

File: Dockerfile [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
-RUN pip install --upgrade pip && pip install pipenv
+RUN pip install --upgrade pip && pip install pipenv==v2022.11.30
```
Comment (gtempus) [inline Dockerfile:9]: Why did we have to pin `pipenv`?

**Review by gtempus — CHANGES_REQUESTED** [review]
Thanks, @dmannarino! I just had a couple of questions about the dependencies.

File: Dockerfile (hunk 1 above)
Comment (dmannarino) [reply Dockerfile:7]: Building the image failed due to one of the Python packages failing to compile, complaining that make couldn't be found. Adding this fixed it. It seems that the newer Docker image has different packages included by default (I think it uses a newer version of Debian), which is no big deal.

Comment (gtempus) [reply Dockerfile:7]: Gotcha. Yeah, I thought we talked about this before with another image. I had deja vu when posting the question. 😆 
Thanks!

File: Dockerfile (hunk 2 above)
Comment (dmannarino) [reply Dockerfile:9]: The version of pipenv from the day before had a bug that prevented the lockfile being generated, so when I finally got it to work when the new version came out I decided to pin it. pipenv just doesn't have a great track record WRT bugs, so I decided pinning it is better than having builds start failing at some point in the future.

**Review by gtempus — APPROVED** [review]

**Pull request merged**